homeassistant/components/epson/__init__.py

```diff
@@ -1 +1,56 @@
-"""The epson component."""
+"""The epson integration."""
+import asyncio
+
+from homeassistant.components.media_player import DOMAIN as MEDIA_PLAYER_PLATFROM
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import DATA_EPSON, DOMAIN, SIGNAL_CONFIG_OPTIONS_UPDATE, UPDATE_LISTENER
+
+PLATFORMS = [MEDIA_PLAYER_PLATFROM]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the epson component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up epson from a config entry."""
+    hass.data[DOMAIN][entry.entry_id] = {
```

Comment:
Let's create the `EpsonProjector` object before storing anything in `hass.data` and test if the connection works. If not, raise `ConfigEntryNotReady`

Reply:
do I assume correctly that you mean `epson.Projector` not `EpsonProjector` which is of type `MediaPlayerEntity`?

Reply:
Yes, sorry.